apps/sms/js/thread_list_ui.js

```diff
@@ -449,88 +455,80 @@ var ThreadListUI = {
+      // other platform APIs expect this id to be a number.
+      var threadId = +id;
+      var isDraft = typeof Threads.get(threadId) === 'undefined';
+
+      if (isDraft) {
+        draftIds.push(threadId);
+        this.removeThread(threadId);
+      } else {
+        threadIds.push(threadId);
+        this.deleteThread(threadId);
       }
+    });
 
-      WaitingScreen.show();
-
-      threadIdsToDelete = selected.reduce(function(list, value) {
-        // Coerce the threadId back to a number MobileMessageFilter and all
-        // other platform APIs expect this value to be a number.
-        var threadId = +value;
-        var isDraft = typeof Threads.get(threadId) === 'undefined';
-
-        if (isDraft) {
-          Drafts.delete(Drafts.get(threadId));
-          ThreadListUI.removeThread(threadId);
-        } else {
-          list.push(threadId);
-        }
+    this.performDeleteThreadDraft(selected, draftIds, threadIds);
+  },
 
-        return list;
-      }, []);
+  performDeleteThreadDraft:
+    function thlui_performDeleteThreadDraft(selected, draftIds, threadIds) {
+    var threadCountToDelete = 0, messageIdsToDelete = [];
 
-      // That means that we've just removed some drafts
-      if (threadIdsToDelete.length !== selected.length) {
-        Drafts.store();
-      }
+    if(draftIds) {
+      draftIds.forEach(function(threadId) {
+        Drafts.delete(Drafts.get(threadId));
+      });
+    }
+    if (threadIds.length !== selected.length) {
+      Drafts.store();
+    }
 
-      if (!threadIdsToDelete.length) {
-        exitEditMode();
-        return;
+    var onAllThreadMessagesRetrieved = () => {
+      if (!--threadCountToDelete) {
+        MessageManager.deleteMessages(messageIdsToDelete);
+        this.cancelEdit();
       }
+    };
 
-      threadCountToDelete = threadIdsToDelete.length;
+    var onThreadMessageRetrieved = (message) => {
+      messageIdsToDelete.push(message.id);
+      return true;
+    };
 
-      threadIdsToDelete.forEach(function(threadId) {
-        MessageManager.getMessages({
-          // Filter and request all messages with this threadId
-          filter: { threadId: threadId },
-          each: onThreadMessageRetrieved,
-          end: onAllThreadMessagesRetrieved
-        });
+    threadCountToDelete = threadIds.length;
+    threadIds.forEach(function(threadId) {
+      MessageManager.getMessages({
+        // Filter and request all messages with this threadId
+        filter: { threadId: threadId },
+        each: onThreadMessageRetrieved,
+        end: onAllThreadMessagesRetrieved
       });
-    }
+    });
+  },
 
-    return Utils.confirm(
-      {
-        id: 'deleteThreads-confirmation-message',
-        args: { n: selected.length }
-      },
-      null,
-      {
-        text: 'delete',
-        className: 'danger'
-      }
-    ).then(performDeletion.bind(this));
+  exitEditMode: function thlui_exitEditMode() {
+    this.cancelEdit();
```

Comment:
and actually that's what you do already.. so you can just remove this function :)